specification/compute/resource-manager/Microsoft.Compute/stable/2019-11-01/disk.json

```diff
@@ -0,0 +1,2030 @@
+        },
+        "encryption": {
+          "$ref": "#/definitions/Encryption",
+          "description": "Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys."
+        }
+      },
+      "description": "Snapshot resource update properties."
+    },
+    "DiskEncryptionSetUpdateProperties": {
+      "properties": {
+        "activeKey": {
+          "$ref": "#/definitions/KeyVaultAndKeyReference"
+        }
+      },
+      "description": "disk encryption set resource update properties."
+    },
+    "CreationData": {
+      "properties": {
+        "createOption": {
+          "type": "string",
+          "enum": [
+            "Empty",
+            "Attach",
+            "FromImage",
+            "Import",
+            "Copy",
+            "Restore",
+            "Upload"
+          ],
+          "x-ms-enum": {
+            "name": "DiskCreateOption",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Empty",
+                "description": "Create an empty data disk of a size given by diskSizeGB."
+              },
+              {
+                "value": "Attach",
+                "description": "Disk will be attached to a VM."
+              },
+              {
+                "value": "FromImage",
+                "description": "Create a new disk from a platform image specified by the given imageReference or galleryImageReference."
+              },
+              {
+                "value": "Import",
+                "description": "Create a disk by importing from a blob specified by a sourceUri in a storage account specified by storageAccountId."
+              },
+              {
+                "value": "Copy",
+                "description": "Create a new disk or snapshot by copying from a disk or snapshot specified by the given sourceResourceId."
+              },
+              {
+                "value": "Restore",
+                "description": "Create a new disk by copying from a backup recovery point."
+              },
+              {
+                "value": "Upload",
+                "description": "Create a new disk by obtaining a write token and using it to directly upload the contents of the disk."
+              }
+            ]
+          },
+          "description": "This enumerates the possible sources of a disk's creation."
+        },
+        "storageAccountId": {
+          "type": "string",
+          "description": "Required if createOption is Import. The Azure Resource Manager identifier of the storage account containing the blob to import as a disk."
+        },
+        "imageReference": {
+          "$ref": "#/definitions/ImageDiskReference",
+          "description": "Disk source information."
+        },
+        "galleryImageReference": {
+          "$ref": "#/definitions/ImageDiskReference",
+          "description": "Required if creating from a Gallery Image"
```

Comment:
Add: "ARM id of the shared galley image version from which to create a disk"

Reply:
added

Reply:
I'll add 'Required if creating from a Gallery Image. The id of the ImageDiskReference will be the ARM id of the shared galley image version from which to create a disk.' since this object isn't a URI or string,